frame/generic-asset/src/impls.rs

```
@@ -0,0 +1,594 @@
+	fn multi_accounting_withdraw_reserved_funds_should_fail() {
+		let (alice, asset_id, amount) = (1, 16000, 100);
+		ExtBuilder::default()
+			.free_balance((asset_id, alice, amount))
+			.build()
+			.execute_with(|| {
+				// Lock alice's funds
+				GenericAsset::reserve(&asset_id, &alice, amount).ok();
+
+				assert_noop!(
+					<GenericAsset as MultiCurrencyAccounting>::withdraw(
+						&alice,
+						Some(asset_id),
+						amount,
+						WithdrawReasons::all(),
+						ExistenceRequirement::KeepAlive
+					),
+					"account has too few funds"
+				);
+			});
+	}
+
+	#[test]
+	fn multi_accounting_make_free_balance_edge_cases() {
+		let (alice, asset_id) = (&1, 16000);
+		ExtBuilder::default()
+			.build()
+			.execute_with(|| {
+				let max_value = u64::max_value();
+				let min_value = Zero::zero();
+
+				let _ = <GenericAsset as MultiCurrencyAccounting>::make_free_balance_be(
+					alice,
+					Some(asset_id),
+					max_value,
+				);
+				// Check balance updated
+				assert_eq!(GenericAsset::total_issuance(asset_id), max_value);
+				assert_eq!(
+					<GenericAsset as MultiCurrencyAccounting>::free_balance(alice, Some(asset_id)),
+					max_value
+				);
+
+				let _ = <GenericAsset as MultiCurrencyAccounting>::make_free_balance_be(
+					alice,
+					Some(asset_id),
+					min_value,
+				);
+				// Check balance updated
+				assert_eq!(GenericAsset::total_issuance(asset_id), min_value);
+				assert_eq!(
+					<GenericAsset as MultiCurrencyAccounting>::free_balance(alice, Some(asset_id)),
+					min_value
+				);
+			});
+	}
+
+	#[test]
+	#[ignore]
+	#[should_panic(expected = "attempt to add with overflow")]
+	fn multi_accounting_deposit_creating_overflow_should_panic() {
+		let (alice, asset_id, amount) = (1, 16000, 100);
+		ExtBuilder::default()
+			.build()
+			.execute_with(|| {
+				let max_value = u64::max_value();
+				let _ = <GenericAsset as MultiCurrencyAccounting>::deposit_creating(
+					&alice, 
+					Some(asset_id), 
+					max_value + amount
```

In this test overflow happens here before the (`::deposit_creating`) is called?
Can remove this test. Also nothing should panic in the runtime, it should return an error

---

Will remove this one